libbpf-tools/profile.c

```diff
@@ -270,26 +271,29 @@ static int cmp_counts(const void *a, const void *b)
 
 static int read_counts_map(int fd, struct key_ext_t *items, __u32 *count)
 {
+	struct key_t keys[*count];
+	__u64 vals[*count];
+	__u32 key_size = sizeof(keys[0]);
+	__u32 value_size = sizeof(__u64);
 	struct key_t empty = {};
-	struct key_t *lookup_key = &empty;
 	int i = 0;
 	int err;
+	__u32 n = *count;
 
-	while (bpf_map_get_next_key(fd, lookup_key, &items[i].k) == 0) {
-		err = bpf_map_lookup_elem(fd, &items[i].k, &items[i].v);
-		if (err < 0) {
-			fprintf(stderr, "failed to lookup counts: %d\n", err);
-			return -err;
-		}
-
-		if (items[i].v == 0)
-			continue;
+	err = dump_hash(fd, keys, key_size, vals, value_size, &n, &empty);
+	if (err)
+		return err;
 
-		lookup_key = &items[i].k;
-		i++;
+	for (i = 0; i < n; i++) {
```

Comment:
In previous patches, maps were read directly into the "items" variable, so this copy would not work unless it was in a batch. Now we use a little more memory and CPU. However, this change is good for eliminating duplicate code.